src/tensor_ops/utils.rs

```diff
@@ -38,7 +38,10 @@ where
 {
     T::Device::foreach_m(t.mut_data(), &mut |x| *x = f(x)); // clones if there is more than 1 reference to t
     let (t, mut tape) = t.split_tape();
-    let result = t.clone(); // will always a new reference to t, not start a new one
+
+    let mut result: T::NoTape = TensorCreator::zeros();
+    result.mut_data().clone_from(t.data());
```

Comment:
Is it possible to do this with re-allocating every time? this function was first introduced because it could reduce allocations for certain things, like softmax:
```rust
x.log_softmax().exp()
```

Here only 1 thing will be referencing `x.log_softmax()`, so the reference count will be 1. That means we can just re-use the memory instead of allocating new data.